### PR TITLE
Initial Offline Strawman

### PR DESCRIPTION
**Pull request by secondsun**
Motivation:
This is some initial documentation HEAVILY influenced by some preliminary Android work.  The PR should be mostly for discussion and iteration of this idea.

### CODE REVIEW

File: docs/specs/aerogear-offline/index.markdown [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# API
+
+## Implementation 
+
+### Storage API
+ * `get` : *URL* **url**, *Callback* **callback**   
+This should check and see if a file related to the URL is available. 
+     * If the file is locally stored, the file should be passed to the callback's success method. 
+     * If the file is not it should be fetched, saved, and returned to the callback's success method.
```
Comment (matzew) [inline docs/specs/aerogear-offline/index.markdown]: Just wondering: Is the idea to also leverage Expires HTTP headers, if present, to see if the local doc is still "up to date" ?

Comment (secondsun) [inline docs/specs/aerogear-offline/index.markdown]: URL's aren't necessarily HTTP ;) put yes there will be a goal to add metadata to manage a file expiring.

Comment (matzew) [inline docs/specs/aerogear-offline/index.markdown]: cool!

On Thu, Apr 16, 2015 at 3:58 PM, Hoyt Summers Pittman <
notifications@github.com> wrote:

> In docs/specs/aerogear-offline/index.markdown
> <https://github.com/aerogear/aerogear.org/pull/573#discussion_r28511415>:
>
> > +<span class="label label-warning">Status: Experimental</span>
> > +
> > +# Goals
> > +
> > +1. To manage resources (assumed files) on the mobile devices which are retrieved from transitory sources (ie online, SD cards, etc).
> > +2. To provide a way to secure, and expire offline resource. (ie encryption and TTL policies)
> > +
> > +# API
> > +
> > +## Implementation
> > +
> > +### Storage API
> > + * `get` : *URL* **url**, *Callback* **callback**
> > +This should check and see if a file related to the URL is available.
> > +     * If the file is locally stored, the file should be passed to the callback's success method.
> > +     * If the file is not it should be fetched, saved, and returned to the callback's success method.
>
> URL's aren't necessarily HTTP ;) put yes there will be a goal to add
> metadata to manage a file expiring.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear.org/pull/573/files#r28511415>.
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Comment by danielpassos** [comment]
:+1: I think it's a good start

**Comment by abstractj** [comment]
@secondsun I wonder if we should merge it against another branch, like `offline_spec`. Then when we think it's ready to go for production, merge into master. Wdyt?

**Comment by secondsun** [comment]
@abstractj  see https://github.com/aerogear/aerogear.org/pull/576 :)

**Pull request closed without merging**